### ISSUE TITLE
Unresolved import with `macro_rules_attribute`

### ISSUE THREAD

**Issue by ModProg** [rust-lang/rust-analyzer#14117]
**rust-analyzer version**: `rust-analyzer 1.69.0-nightly (8996ea9 2023-02-09)`

**rustc version**:  `rustc 1.67.1 (d5a82bbd2 2023-02-07)`

The following code runs/compiles fine, but in rust-analyzer yields an `unresoved import` error on `use module::Struct`, that seems to be caused by `apply`. Importing from the parent module works without error (i.e. `use crate::NonModule` does not produce any error).
```rs
#[macro_use]
extern crate macro_rules_attribute;

attribute_alias! {
    #[apply(Test)] = #[derive(Debug)];
}

mod module {
    use crate::NonModule;

    #[apply(Test)]
    pub struct Struct(pub NonModule);
}

use module::Struct;

#[apply(Test)]
pub struct NonModule;

fn main() {
    println!("{:?}", Struct(NonModule));
    println!("{NonModule:?}");
}
```

**Comment by lnicola** [comment]
Do we expand that `attribute_alias!` call?

**Comment by ModProg** [comment]
> Do we expand that `attribute_alias!` call (press F1, there's an `Expand macro` command)?

Yes:
```rs
// ================================================
// Recursive expansion of the attribute_alias macro
// ================================================

macro_rules! __emit__ {
    ($_:tt) => {
        #[allow(nonstandard_style)]
        macro_rules!Test {
            ($_($item:tt)*) => (#[derive(Debug)]$_($item)*)
          }
        #[allow(unused_imports)]
        pub(crate) use Test;
    };
}
#[allow(nonstandard_style)]
macro_rules! Test {
  ($($item:tt)*) => (#[derive(Debug)]$($item)*)
}
#[allow(unused_imports)]
pub(crate) use Test;

```

The apply one as well:
```rs
Test! {
  pub struct Struct(pub NonModule);
}
```
